### ISSUE TITLE
Clean up expired bindings

### ISSUE THREAD

**Issue by lnrdll** [GoogleCloudPlatform/jit-access#150]
While it might seem like a long shot, this issue primarily stems from the API and not this tool. Nevertheless, it's valuable to engage in a discussion regarding this matter and explore potential solutions.

When a temporary IAM binding is established, it appears in GCP's IAM view, complete with the condition name in the 'Conditions' column. The underlying 'problem' arises when these temporary bindings expire; they persist in the IAM view, potentially causing the list to grow significantly. Here's an example of an expired binding that remains visible in the IAM list:

![Screenshot 2023-09-06 4 39 14 PM](https://github.com/GoogleCloudPlatform/jit-access/assets/25017746/658cce7c-fa00-4332-90ce-89fca84851bc)

I recall from a different issue thread that the intentional absence of a database was a deliberate design choice, driven by security considerations. A database would simplify the cleanup process. However, even though this issue is primarily related to the API and not the JIT tool, is there a way to address it within the application itself?

- Could the application, for instance, store a local file to keep track of projects requiring cleanup?
- Or could it publish project IDs to a Pub/Sub, allowing a background job to check for expired bindings?
- Another possibility is for the application to retain this information in memory.

**Comment by jpassing** [comment]
You're right that expired IAM bindings cause some clutter. FWIW, the application does [purge expired bindings](https://github.com/GoogleCloudPlatform/jit-access/blob/5fffbac99c76f16205c69fc3b2abd6aeb6f87e4c/sources/src/main/java/com/google/solutions/jitaccess/core/adapters/ResourceManagerAdapter.java#L132) if a user requests the same role again -- so there's a limit to how many of these expired role bindings you can possibly accumulate in an IAM policy. 

> Or could it publish project IDs to a Pub/Sub, allowing a background job to check for expired bindings?

There [was a PR attempting to do just that](https://github.com/GoogleCloudPlatform/jit-access/pull/97), but I'd have to follow up what the status is.

**Comment by jpassing** [comment]
> There https://github.com/GoogleCloudPlatform/jit-access/pull/97, but I'd have to follow up what the status is.

#154 will add Pub/Sub integration

**Comment by eeeeethan2333** [comment]
> > There #97, but I'd have to follow up what the status is.
> 
> #154 will add Pub/Sub integration

And after this pub/sub integration, I'm planning to provide pubsub consumer example to clean up the roles.

**Comment by jpassing** [comment]
JIT Groups no longer uses temporary role bindings, so I believe this issue is obsolete now
